### PR TITLE
[NET-6417] Fix meshgw tests

### PR DESCRIPTION
**Pull request by jm96441n**
### Changes proposed in this PR ###  
- fixes tests for meshgw that were broken by adding https://github.com/hashicorp/consul/pull/20425 validation

### How I've tested this PR ###
ran tests

### Checklist ###
- [X] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: control-plane/controllers/resources/mesh_gateway_controller_test.go [hunk 1]
```diff
@@ -444,7 +586,7 @@ func TestMeshGatewayController_Reconcile(t *testing.T) {
 
 			res, err := controller.Reconcile(context.Background(), testCase.request)
 			if testCase.expectedErr != nil {
-				//require.EqualError(t, err, testCase.expectedErr.Error())
+				// require.EqualError(t, err, testCase.expectedErr.Error())
```
Comment (nathancoleman) [inline control-plane/controllers/resources/mesh_gateway_controller_test.go:589]: Should this be here? Out of scope for this PR, just curious

File: control-plane/controllers/resources/mesh_gateway_controller_test.go [hunk 2]
```diff
@@ -50,6 +52,16 @@ func TestMeshGatewayController_Reconcile(t *testing.T) {
 						Namespace: "consul",
 						Name:      "mesh-gateway",
 					},
+					Spec: pbmesh.MeshGateway{
+						GatewayClassName: "consul",
+						Listeners: []*pbmesh.MeshGatewayListener{
+							{
+								Name:     "wan",
+								Port:     8443,
+								Protocol: "mesh",
```
Comment (nathancoleman) [inline control-plane/controllers/resources/mesh_gateway_controller_test.go:61]: This is currently required to be `TCP` [here](https://github.com/hashicorp/consul/blob/6a8554317502dc54fb794ce950db8678ec352c55/proto-public/pbmesh/v2beta1/mesh_gateway.proto#L29-L30). Will be explicitly `mesh` in the future; however, I'd make it TCP for now in these tests so as not to run into any surprises.

File: control-plane/controllers/resources/mesh_gateway_controller_test.go (hunk 1 above)
Comment (jm96441n) [reply control-plane/controllers/resources/mesh_gateway_controller_test.go:589]: honestly I don't know, this was gofumpt editing the spacing on the comment

**Review by nathancoleman — APPROVED** [review]
Thanks for picking up and fixing this issue so quickly! 🥇

**Pull request merged**